launcher/InstanceImportTask.cpp

```diff
@@ -735,13 +740,24 @@ void InstanceImportTask::processModrinth()
     instance.saveNow();
 
     m_filesNetJob = new NetJob(tr("Mod download"), APPLICATION->network());
-    for (auto &file : files)
+    for (auto file : files)
     {
         auto path = FS::PathCombine(m_stagingPath, ".minecraft", file.path);
-        qDebug() << "Will download" << file.download << "to" << path;
-        auto dl = Net::Download::makeFile(file.download, path);
+        qDebug() << "Will try to download" << file.downloads.front() << "to" << path;
+        auto dl = Net::Download::makeFile(file.downloads.front(), path);
         dl->addValidator(new Net::ChecksumValidator(file.hashAlgorithm, file.hash));
         m_filesNetJob->addNetAction(dl);
+
+        if (file.downloads.size() > 1) {
+            // FIXME: This really needs to be put into a ConcurrentTask of
+            // MultipleOptionsTask's , once those exist :)
+            connect(dl.get(), &NetAction::failed, [this, &file, path, dl]{
+                auto dl = Net::Download::makeFile(file.downloads.dequeue(), path);
```

Comment:
won't this use the same URL as before. We haven't dequeued it before.

Reply:
oops T-T